diff --git a/sfeed_update b/sfeed_update
index bc74470..4d4d476 100755
--- a/sfeed_update
+++ b/sfeed_update
@@ -239,10 +239,9 @@ main() {
 # process a single feed.
 # parameters are: config, tmpdir, name, feedurl, basesiteurl, encoding
 if [ "${SFEED_UPDATE_CHILD}" = "1" ]; then
-	IFS="" # "\037"
 	[ "$1" = "" ] && exit 0 # must have an argument set
 	printf '%s\n' "$1" | \
-	while read -r _config _tmpdir _name _feedurl _basesiteurl _encoding; do
+	while IFS="" read -r _config _tmpdir _name _feedurl _basesiteurl _encoding; do
 		loadconfig "${_config}"
 		sfeedtmpdir="${_tmpdir}"
 		_feed "${_name}" "${_feedurl}" "${_basesiteurl}" "${_encoding}"
